cargo/src/bin/cargo_audit.rs

```diff
@@ -24,6 +24,75 @@
 #![deny(clippy::manual_let_else)]
 #![allow(clippy::unreachable)]
 
-fn main() {
-    println!("Todo-uwu");
+use obs_service_cargo::audit;
+use obs_service_cargo::audit::Src;
+use std::error::Error;
+use std::fmt;
+use std::fmt::Debug;
+use std::fmt::Display;
+
+struct AuditError {
+    msg: String,
+}
+
+impl Debug for AuditError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self.msg)
+    }
+}
+
+impl Display for AuditError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self.msg)
+    }
+}
+
+impl Error for AuditError {}
+
+fn main() -> Result<(), AuditError> {
+    if let Some(arg) = std::env::args().nth(1) {
+        let pathtofile = std::path::PathBuf::from(arg);
+        if let Ok(this) = audit::read_service(pathtofile.as_path()) {
+            // TODO: Use the values to find Cargo.lock
```

Comment:
Could this be shared from the cargo-vendor? Cargo.lock tends to end up either at the root (workspace) or next to the vendor dir. So we could re-use that logic ...

Reply:
true. i will resume this branch here after i refactor a lot of stuff